src/lib.rs

```diff
@@ -393,10 +402,15 @@ impl<'a> ModalityProbe<'a> {
     pub fn new_with_storage(
         history_memory: &'a mut [u8],
         probe_id: ProbeId,
+        restart_counter: RestartCounterProvider<'a>,
     ) -> Result<ModalityProbe<'a>, StorageSetupError> {
-        let t = ModalityProbe::<'a> {
-            history: DynamicHistory::new_at(history_memory, probe_id)?,
+        let mut t = ModalityProbe::<'a> {
+            history: DynamicHistory::new_at(history_memory, probe_id, restart_counter)?,
         };
+        t.record_event_with_payload(
+            EventId::EVENT_PROBE_INITIALIZED,
+            t.history.probe_id.get_raw(),
```

Comment:
As noted in `src/id.rs`, I think the payload is redundant, using log-space for no value.